corehq/apps/app_manager/static/app_manager/js/details/screen_config.js

```diff
@@ -789,36 +789,33 @@ hqDefine('app_manager/js/details/screen_config', function () {
                 });
 
                 self.save = function () {
-                    var i;
-                    //Only save if property names are valid
-                    var containsTab = false;
-                    var columns = self.columns();
-                    for (i = 0; i < columns.length; i++) {
-                        var column = columns[i];
+                    // Only save if property names are valid
+                    var errors = [],
+                        containsTab = false;
+                    _.each(self.columns(), function (column) {
                         column.saveAttempted(true);
-                        if (!column.isTab) {
-                            if (column.showWarning()) {
-                                alert(gettext("There are errors in your property names"));
-                                return;
-                            }
-                        } else {
+                        if (column.isTab) {
+                            containsTab = true;
                             if (column.showWarning()) {
-                                alert(gettext("There are errors in your tabs"));
-                                return;
+                                errors.push(gettext("There is an error in your tab: ") + column.field.value);
                             }
-                            containsTab = true;
+                        } else if (column.showWarning()) {
+                            errors.push(gettext("There is an error in your property name: ") + column.field.value);
                         }
-                    }
+                    });
                     if (containsTab) {
-                        if (!columns[0].isTab) {
-                            alert(gettext("All properties must be below a tab"));
-                            return;
+                        if (!self.columns()[0].isTab) {
+                            errors.push(gettext("All properties must be below a tab."));
                         }
                     }
+                    if (errors.length) {
+                        alert(gettext("There are errors in your configuration.\n") + errors.join("\n"));
```

Comment:
I think you should move the `\n` out of the `gettext` as this could be removed in translations accidentally and should always be there.

Reply:
indeed, 83fc3cf17a295353c57e9e0edd8877728c73ec5f